R/zchunk_LA154.transportation_UCD.R

```diff
@@ -40,120 +44,390 @@ module_energy_LA154.transportation_UCD_DISABLED <- function(command, ...) {
+
+    # NOTE: We are currently aggregating IEA's data on rail and road due to inconsistencies (e.g. no rail in the Middle East)
+    # First, replace the international shipping data (swapping in EIA for IEA)
+    # Only perform this swap for international shipping / refined liquids, and in countries in the EIA database
+    IEA_data_EIA_intlship <- L101.in_EJ_ctry_trn_Fi_Yh %>%
+      # expecting NAs here because we only want to replace certain values
+      left_join_keep_first_only(L1011.in_EJ_ctry_intlship_TOT_Yh %>% rename(EIA_value = value), by = c("iso","year")) %>%
+      mutate(value = if_else(sector == "in_trn_international ship" &
+                               fuel == "refined liquids" &
+                               !is.na(EIA_value), EIA_value, value),
+             sector = sub("in_", "", sector)) %>%
+      select(iso, sector, fuel, year, value)
+
+    # Need to map IEA sector to UCD_category, calibrated_techs_trn_agg data is too busy
+    UCD_category_mapping <- calibrated_techs_trn_agg %>% select(sector, UCD_category) %>% distinct
+
+    # Aggregate IEA data to UCD_category in each country/year instead of sector
+    IEA_data_aggregated_by_UCD_cat <- IEA_data_EIA_intlship %>%
+      left_join_error_no_match(UCD_category_mapping, by = "sector") %>%
+      group_by(iso, UCD_category, fuel, year) %>%
+      summarise(value = sum(value))
+
+    # Aggregating UCD transportation database by the general categories used for the IEA transportation data
+    # These will be used to compute shares for allocation of energy to mode/technology/fuel within category/fuel
+    UCD_trn_data_UCD_techs <- UCD_trn_data %>%
+      filter(variable == "energy") %>%
+      left_join_error_no_match(UCD_techs, by = c("UCD_sector", "mode", "size.class", "UCD_technology", "UCD_fuel"))
+
+    UCD_trn_data_UCD_cat <- UCD_trn_data_UCD_techs %>%
+      # Filtering only to base year for computing shares
+      filter(year == energy.UCD_EN_YEAR) %>%
+      group_by(UCD_region, UCD_category, fuel) %>%
+      summarise(agg = sum(value))
+
+    # Match these energy quantities back into the complete table for computation of shares of fuel in category
+    UCD_trn_data_UCD_techs <- UCD_trn_data_UCD_techs %>%
+      filter(year == energy.UCD_EN_YEAR) %>%
+      left_join_error_no_match(UCD_trn_data_UCD_cat, by = c("UCD_region", "UCD_category", "fuel")) %>%
+      # If the aggregate is 0 or value is NA, set share to 0, rather than NA
+      mutate(UCD_share = if_else(agg == 0 | is.na(value), 0, value/agg))
+
+    # Writing out the UC Davis mode/technology/fuel shares within category/fuel at the country level
+    # First, creating a table of desired countries with their UCD regions
+    ctry_iso_region <- tibble(iso = unique(L101.in_EJ_ctry_trn_Fi_Yh$iso)) %>%
+      left_join_error_no_match(UCD_ctry, by = "iso")
+
+    UCD_fuel_share_in_cat <- UCD_trn_data_UCD_techs %>%
+      repeat_add_columns(ctry_iso_region) %>%
+      filter(UCD_region.x == UCD_region.y) %>%
```

Comment:
I'm confused where the .x and .y versions came from. Maybe a comment would help.